src/logic/calculate.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable no-unused-vars */
+import { operate } from './operate';
+
+export const calculate = (calculator, button) => {
+  let { total, next, operation } = calculator;
+
+  if (button === '+/-') {
```

Comment:
- [ ] Good job done here, but we need to handle more cases to cover all the possible scenarios.
Here are some cases to consider to structure the calculator object `total, next, operation`:
- what if the button is a number?
- what if the button is a number and the previous click is an operation?
- what if the button is '.'

Kindly make sure to cover all the possible scenarios.